### DISCUSSION TITLE
[A/L] (2021/22) Combinatoria - Esercizio 3

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#179] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152252222-d277914b-36f8-4bcd-9a2c-a5b8c2fe6b16.png)

**Comment by Exyss** [comment]
1. 10! 
2. $\dfrac{10!}{7!}$
3. $\binom{10}{5}$
4. $3 \cdot 9!$
5. $9! \cdot 3$

6. Immaginiamo che Tizio sia in prima posizione. Allora avremmo 9 combinazioni con Caio dopo tizio, poiché i posti di arrivo sotto di Tizio sono 9. Immaginiamo ora che Tizio sia in seconda posizione. Questa volta avremmo 8 combinazioni con Caio dopo tizio. Applicando tale ragionamento fino al far slittare Tizio e Caio fino alle ultime due posizioni, otteniamo $8! \cdot \sum\limits_{i=1}^9 i$.
Le due domande proposte sono equivalenti, dunque basta rispondere ad una sola
![Esercizio 6](https://user-images.githubusercontent.com/65043207/152385389-bb4a3198-43d5-4358-9da3-7771b7caffe7.jpg)

9. La spiegazione di questo caso è analoga alla precedente, tuttavia strutturata su "tre livelli": immaginiamo di far prima slittare Caio e poi di far slittare Tizio. Otteniamo così una "doppia sommatoria innestata" (se può aiutare, si immagini la situazione come se fosse un problema ricorsivo)

![Esercizio 7](https://user-images.githubusercontent.com/65043207/152387499-4ec5bb09-0b4a-44a6-b720-8f00c72108c9.jpg)

**Comment by AxnNxs** [comment]
- Ordini di arrivo possibili:
Si tratta di una permutazione, ovvero tutti i possibili ordinamenti sul totale (n). Poiché n=10, la soluzione è 10! = 3.628.800

- Possibili salite al podio:
L'ordine conta, in quanto ci interessa sapere chi nel gruppo dei primi 3 ha tagliato prima il traguardo. Per il Principio Moltiplicativo, possiamo scegliere prima tra 10, poi tra 9 e poi tra gli 8 atleti rimanenti. Pertanto la soluzione è 10x9x8 = 720.
NB: se ci interessasse soltanto sapere quante sono le possibili salite al podio senza preoccuparci dell'ordine, la soluzione sarebbe 10 scelgo 3, ovvero, secondo la formula di coefficiente binomiale: 10! / (3! x 7!) 

- Possibili gruppi di 5 atleti che si posizionano prima degli altri, passando il turno:
Essendo l'esito booleano, ovvero passato/non passato, non ci interessa l'ordinamento, ci basta osservare le possibili combinazioni di arrivo per 5 atleti al netto delle ripetizioni: come per la nota di prima, utilizzando il coefficiente binomiale, si ottiene che la soluzione è 10! / 5! x 5! = 36

- Ordini di arrivo con un inglese al primo posto:
Sapendo che il primo deve essere inglese e le restanti posizioni possono essere ricoperte da un atleta qualsiasi, avremo 3 scelgo 1 possibilità che il primo sia inglese, poiché in gara ve ne sono 3; i restanti atleti potremo sceglierli senza preferenza tra i 9 rimanenti.
Pertanto la soluzione è 3 x 9! = 1.088.640

- Ordini di arrivo con un inglese all'ultimo posto:
Come prima

- Possibili ordinamenti in cui, presi due elementi da un insieme X, questi rimangono nella stessa relazione "minore/maggiore di". 
Nell'insieme Italiani di 3 elementi, abbiamo T, C e S. Fissiamo T per 9 posizioni (non ha senso metterlo in ultima posizione, se deve sempre piazzarsi prima di C). Man mano però che T si piazza in posizioni inferiori, C avrà una possibilità in meno, di volta in volta, per potersi piazzare dietro a T. Pertanto:
9 possibili posizionamenti di T, 9 possibili iniziali posizionamenti per C, per poi scendere progressivamente. La soluzione è:
9 x 9! = 3.265.920 (da verificare).

**Comment by GattoBrizzolato** [comment]
1. Se contiamo tutti gli atleti come distinti allora la soluzione è 10! MA il problema sembra posto diversamente : presa una posizione x a noi interessa sapere la nazionalità non l'individuo. ESEMPIO : presa una parola come 'CINA' sappiamo che tutti i possibili anagrammi (contando anche i no sense) sono 4! ma se prendiamo la parola 'NONO' notiamo che non possono esserci 4! soluzioni poiché così facendo si andrebbero a contare anche le ripetizioni. 'NNOO' ad esempio ripeterebbe per 
( 2! x 2! ) volte lo stesso anagramma. La soluzione del problema originale diventa quindi :  10! / (3! x 4! x 3! )

6) La soluzione più semplice è  10! / 2 : semplicemente in metà delle combinazioni Tizio sarà sempre prima di Caio e viceversa nell'altra metà

**Comment by Exyss** [comment]
@GattoBrizzolato 
- **Riguardo al punto 1:** approvo il ragionamento, ma credo che non sia posto diversamente il problema poiché la domanda chiede "gli ordini di arrivo possibili" e non "gli ordini di arrivo possibili considerando la nazionalità", quindi credo vengano chiesti semplicemente gli ordini totali di arrivo

- **Riguardo al punto 6:** Wow sono rimasto sorpreso dal vedere che esista una soluzione così semplice (tralaltro **confermata per Doppio Conteggio**, entrambe le nostre soluzioni danno come risultato $1.814.400$). Credi che una soluzione simile alla tua possa trovarsi anche per il punto 7? Forse con una tipizzazione o un PIE che sfrutti il tuo metodo di calcolo?

--------

*EDIT*
- Auto-risposta alla mia ultima osservazione sul punto 6
**Sì, esiste una soluzione alternativa più semplice.**

Per curiosità, ho deciso di calcolare a quanto ammontasse il valore della mia risposta. Una volta calcolata la "sommatoria innestata", otteniamo $120 \cdot 7! = 604.800$, che corrisponde esattamente ad $\dfrac{1}{3}$ del risultato della soluzione alternativa del punto 6.

**Dunque la risposta più semplice e confermata per Doppio Conteggio è: $\dfrac{10!}{6}$**

----

**BONUS**
Viste le due risposte per emtrambi i casi, immagino che a questo punto il pattern sia **$\dfrac{N!}{K!}$**, dove N è il numero di posizioni e K è il numero di concorrenti "vincolati"

Infatti, ipotizzando un caso in cui imponiamo il vincolo in cui i corridori devono rispettare l'ordine A > B > C > D > E > F > G > H > I > J, la formula "generica" trovata sarebbe $\dfrac{10!}{10!} = 1$, dunque una sola combinazione possibile che possa rispettare il vincolo (cosa che viene da sé, poiché abbiamo vincolato tutti i corridori)

**Comment by matypist** [comment]
Soluzione ad un esercizio equivalente al punto 6 tratta dalle dispense dell'Anno Accademico attuale (A.A. 2022-2023):

![image](https://user-images.githubusercontent.com/69389461/214630841-c5760ca1-6a2a-49aa-9afb-6ba938f87bad.png)

Notiamo che la soluzione 1 corrisponde a [quella](https://github.com/sapienzastudentsnetwork/mmi2122/issues/4#issuecomment-1029179553) che aveva dato inizialmente Exyss, mentre la soluzione 3 a [quella](https://github.com/sapienzastudentsnetwork/mmi2122/issues/4#issuecomment-1029504337) semplificata che aveva dato GattoBrizzolato. Inoltre la soluzione 2 spiega come si arriva al pattern [descritto](https://github.com/sapienzastudentsnetwork/mmi2122/issues/4#issuecomment-1029531229) nell'ultimo commento (sezione "BONUS") da @Exyss (e valida anche per il punto 7).
